### PR TITLE
Fiid 94/inversiones progress bar

### PR DESCRIPTION
**Pull request by francoFinniu**
- "Crear widget de progress bar para las inversiones" [FIID-94](https://finniu-tech.atlassian.net/browse/FIID-94)
- "Invesiones Finalizadas Widget" [FIID-98](https://finniu-tech.atlassian.net/browse/FIID-98)
- ProgresBarInProgress , ToValidateInvestment,CompletmentInvestment are added
- All widgets are designed to be mapped and include soles/dollars and a light-dark theme.

Screenshot 2024-07-08 183824
Screenshot 2024-07-08 183838

### CODE REVIEW

File: lib/presentation/screens/catalog/catalog_screen.dart [hunk 1]
```diff
@@ -25,6 +30,40 @@ class CatalogScreen extends HookConsumerWidget {
       body: SingleChildScrollView(
         child: Column(
           children: [
+            const CompletmentInvestment(
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/catalog_screen.dart:33]: Spelling, CompleteInvestment?

File: lib/presentation/screens/catalog/catalog_screen.dart [hunk 2]
```diff
@@ -25,6 +30,40 @@ class CatalogScreen extends HookConsumerWidget {
...
+              height: 5,
+            ),
+            const ToValidateInvestment(
+              amount: 3000,
+              dateFinal: '10/10/2022',
+            ),
+            const SizedBox(
+              height: 5,
+            ),
+            const ProgresBarInProgress(
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/catalog_screen.dart:56]: spellling,  Progress

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    super.key,
+    required this.dateFinal,
+    required this.amount,
+    required this.isReInvestment,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    int backgroudLight = isReInvestment ? 0xffCFC3FF : 0xffD6F6FF;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:20]: spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    int backgroudDark = isReInvestment ? 0xff6749E2 : 0xff08273F;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:21]: spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+class DownloadButton extends ConsumerWidget {
+  const DownloadButton({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    const int butonColorDark = 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:71]: spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+    const int butonColorLight = 0xff0D3A5C;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:72]: spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+  const AmountInvestment({
+    super.key,
+    required this.amount,
+    required this.isReInvestment,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    int amoutColorDark = isReInvestment ? 0xffC1FF79 : 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:143]: amount

File: lib/presentation/screens/catalog/widgets/investment_completed.dart [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+    int amoutColorLight = isReInvestment ? 0xff278B0E : 0xff0D3A5C;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/investment_completed.dart:144]: amount

File: lib/presentation/screens/catalog/widgets/progres_bar_investment.dart [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  final int amount;
+  const AmountInvestment({
+    super.key,
+    required this.amount,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    const int amoutColorDark = 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/progres_bar_investment.dart:106]: spelling

File: lib/presentation/screens/catalog/widgets/progres_bar_investment.dart [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    const int amoutColorLight = 0xff0D3A5C;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/progres_bar_investment.dart:107]: spelling

File: lib/presentation/screens/catalog/widgets/progres_bar_investment.dart [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+    const int dividerAmoutColor = 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/progres_bar_investment.dart:108]: spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+  const ToValidateInvestment({
+    super.key,
+    required this.dateFinal,
+    required this.amount,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    const backgroudLight = 0xffD6F6FF;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/to_validate_investment.dart:19]: spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+    const backgroudDark = 0xff08273F;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/to_validate_investment.dart:20]: spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+  final int amount;
+  const AmountInvestment({
+    super.key,
+    required this.amount,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isDarkMode = ref.watch(settingsNotifierProvider).isDarkMode;
+    const int amoutColorDark = 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/to_validate_investment.dart:102]: spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+    const int amoutColorLight = 0xff0D3A5C;
+    const int dividerAmoutColor = 0xffA2E6FA;
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/widgets/to_validate_investment.dart:104]: spelling

**Review by harolCalzada — CHANGES_REQUESTED** [review]
I recommend downloading assets at 4x and installing a spell checker plugin for VS.

File: lib/presentation/screens/catalog/catalog_screen.dart (hunk 1 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/catalog_screen.dart]: fix spelling

File: lib/presentation/screens/catalog/catalog_screen.dart (hunk 2 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/catalog_screen.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart (hunk 3 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/investment_completed.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart (hunk 4 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/investment_completed.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/investment_completed.dart (hunk 5 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/investment_completed.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart (hunk 12 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/to_validate_investment.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart (hunk 13 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/to_validate_investment.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart (hunk 14 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/to_validate_investment.dart]: fix spelling

File: lib/presentation/screens/catalog/widgets/to_validate_investment.dart (hunk 15 above)
Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/to_validate_investment.dart]: fix spelling

Comment (francoFinniu) [reply lib/presentation/screens/catalog/widgets/to_validate_investment.dart]: fix spelling

File: lib/presentation/screens/catalog/catalog_screen.dart [hunk 16]
```diff
@@ -25,6 +30,40 @@ class CatalogScreen extends HookConsumerWidget {
       body: SingleChildScrollView(
         child: Column(
           children: [
+            const CompetentInvestment(
```
Comment (harolCalzada) [inline lib/presentation/screens/catalog/catalog_screen.dart:33]: Competent or Complete?

**Review by harolCalzada — CHANGES_REQUESTED** [review]

**Review by harolCalzada — APPROVED** [review]

**Pull request merged**